### PR TITLE
Update lock file for Poetry v1.2.0

### PR DESCRIPTION
**Pull request by erikjohnston**
This is just running `poetry lock --no-update` with 1.2.0

### CODE REVIEW

**Comment by erikjohnston** [comment]
Oh nyargh, this fails with poetry 1.1.5:

```
Installing dependencies from lock file

Package operations: 87 installs, 14 updates, 1 removal

  • Removing setuptools (65.3.0)
  • Updating cffi (1.15.1 -> 1.15.0): Failed

  JSONDecodeError

  Expecting value: line 1 column 1 (char 0)

  at /usr/lib/python3.10/json/decoder.py:355 in raw_decode
      351│         """
      352│         try:
      353│             obj, end = self.scan_once(s, idx)
      354│         except StopIteration as err:
    → 355│             raise JSONDecodeError("Expecting value", s, err.value) from None
      356│         return obj, end
      357│ 

  • Installing pyasn1 (0.4.8): Failed

  JSONDecodeError

  Expecting value: line 1 column 1 (char 0)

  at /usr/lib/python3.10/json/decoder.py:355 in raw_decode
      351│         """
      352│         try:
      353│             obj, end = self.scan_once(s, idx)
      354│         except StopIteration as err:
    → 355│             raise JSONDecodeError("Expecting value", s, err.value) from None
      356│         return obj, end
      357│ 

  • Updating pyparsing (3.0.9 -> 3.0.7): Failed

  JSONDecodeError

  Expecting value: line 1 column 1 (char 0)

  at /usr/lib/python3.10/json/decoder.py:355 in raw_decode
      351│         """
      352│         try:
      353│             obj, end = self.scan_once(s, idx)
      354│         except StopIteration as err:
    → 355│             raise JSONDecodeError("Expecting value", s, err.value) from None
      356│         return obj, end
      357│ 

  • Installing typing-extensions (4.1.1): Failed

  JSONDecodeError

  Expecting value: line 1 column 1 (char 0)

  at /usr/lib/python3.10/json/decoder.py:355 in raw_decode
      351│         """
      352│         try:
      353│             obj, end = self.scan_once(s, idx)
      354│         except StopIteration as err:
    → 355│             raise JSONDecodeError("Expecting value", s, err.value) from None
      356│         return obj, end
      357│ 

  • Installing zipp (3.7.0): Failed

  JSONDecodeError

  Expecting value: line 1 column 1 (char 0)

  at /usr/lib/python3.10/json/decoder.py:355 in raw_decode
      351│         """
      352│         try:
      353│             obj, end = self.scan_once(s, idx)
      354│         except StopIteration as err:
    → 355│             raise JSONDecodeError("Expecting value", s, err.value) from None
      356│         return obj, end
      357│ 
```

**Comment by DMRobertson** [comment]
> Oh nyargh, this fails with poetry 1.1.5:

Is there anything stopping us from saying "Synapse requires Poetry 1.2 to manage its dev environment"?

**Comment by erikjohnston** [comment]
I've tried to add words that explain we've upped the min supported version, but I'm not convinced the words make... sense.

File: docs/development/contributing_guide.md [hunk 1]
```diff
@@ -62,6 +62,8 @@ pipx install poetry
 but see poetry's [installation instructions](https://python-poetry.org/docs/#installation)
 for other installation methods.
 
+Synapse requires a poetry with a minimum version of 1.2.0.
```
Comment (reivilibre) [inline docs/development/contributing_guide.md:65]:
```suggestion
Synapse requires Poetry version 1.2.0 or later.
```
maybe these words make more sense? :D

**Review by reivilibre — APPROVED** [review]

**Pull request merged**